packages/react-utils/src/compose/types.ts

```diff
@@ -39,6 +39,12 @@ export interface BaseSlots {
   root: React.ElementType;
 }
 
+export type SlotPropRenderFunction<TProps> = (Component: React.ElementType<TProps>, props: TProps) => React.ReactNode;
+
+export type ObjectSlotProp<TProps extends GenericDictionary> = TProps & {
```

Comment:
As we have already `ShorthandProps` in this file, should we use `ObjectShorthandProps` or `ObjectProps` 😨 

```suggestion
export type ObjectShorthandProps<TProps extends GenericDictionary> = TProps & {
```

Reply:
Yeah, it was a pure copy+paste from `react-compose`

Reply:
Rename to `ObjectShorthandProps` and removed `SlotPropRenderFunction` since it was a copy of `ShorthandRenderFunction`